src/jaxsim/rbda/soft_contacts.py

```diff
@@ -29,6 +29,23 @@ class SoftContactsParams(JaxsimDataclass):
         default_factory=lambda: jnp.array(0.5, dtype=float)
     )
 
+    def __hash__(self) -> int:
+
+        return hash(
+            (
+                hash(tuple(jnp.atleast_1d(self.K).flatten().tolist())),
+                hash(tuple(jnp.atleast_1d(self.D).flatten().tolist())),
+                hash(tuple(jnp.atleast_1d(self.mu).flatten().tolist())),
```

Comment:
Same as above, is the `.flatten()` necessary?
```suggestion
                hash(tuple(jnp.atleast_1d(self.K).tolist())),
                hash(tuple(jnp.atleast_1d(self.D).tolist())),
                hash(tuple(jnp.atleast_1d(self.mu).tolist())),
```

Reply:
The operator `jnp.atleast1d(X).flatten()` is there mainly for making the hash function work on vectorized pytrees.